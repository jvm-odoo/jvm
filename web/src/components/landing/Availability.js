import React from 'react'

import {
  AvailabilityContainer,
  AvailabilityImage,
  AvailabilityText
} from './'

export const Availability = () => {
  return (
    <AvailabilityContainer>
      <AvailabilityImage />
      <AvailabilityText>
        <span>Available</span> for charity organization project
      </AvailabilityText>
    </AvailabilityContainer>
  )
}