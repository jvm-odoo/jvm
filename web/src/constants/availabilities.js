export const AVAILABILITIES = {
  available: 1,
  partially_available: 2,
  not_available: 3,
}