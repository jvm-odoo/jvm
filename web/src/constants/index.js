export { AVAILABILITIES } from './availabilities'
export { BACKEND_URL } from './server'
export { THEME_LIGHT } from './themes'
