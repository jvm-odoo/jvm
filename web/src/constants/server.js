export const BACKEND_URL = 'https://jvnmhub.xyz'
