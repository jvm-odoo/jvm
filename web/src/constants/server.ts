export const BACKEND_URL = 'http://localhost:3000'
// http://localhost:8080
// https://jvnmhub.xyz
