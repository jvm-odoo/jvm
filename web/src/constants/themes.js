const THEME_LIGHT = {
  mode: 'light',
  background: '#FEFEFE',
  headerBorder: '#EEE',
  title: {
    default: '#24292E',
    inverse: '#FEFEFE',
  },
  text: '#525252',
  wavyContainer: 'linear-gradient(135deg, rgba(90,81,250,0.5) 0%, rgba(179,204,210,0.5) 100%)',
  themeSwitcherShadow: '0px 0px 25px 0px rgba(0,0,0,0.15)',
  inputBackground: '#F5F5F5',
  inputForeground: '#212121',
  colorPrimary: '#6961FB',
}

const THEME_DARK = {
  mode: 'dark',
  background: '#212121',
  headerBorder: '#2f2f2f',
  title: {
    default: '#B5B7FB',
    inverse: '#212121',
  },
  text: '#B1B2CC',
  wavyContainer: 'linear-gradient(135deg, rgba(90,81,250,0.75) 0%, rgba(179,204,210,0.75) 100%)',
  themeSwitcherShadow: '0px 0px 25px 0px #b5b7fb47',
  inputBackground: '#2B2B2B',
  inputForeground: '#FEFEFE',
  colorPrimary: '#B5B7FB',
}

export const themes = {
  'THEME_LIGHT': THEME_LIGHT,
  'THEME_DARK': THEME_DARK,
}
