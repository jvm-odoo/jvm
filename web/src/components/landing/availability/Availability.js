import React, { useEffect } from 'react'
import { useQuery } from '@apollo/react-hooks'
import { gql } from 'apollo-boost'
import { useSelector, useDispatch } from 'react-redux'

import {
  AvailabilityContainer,
  AvailabilityImage,
  AvailabilityText
} from '../'
import { AVAILABILITIES } from '../../../constants'
import { setAvailability } from '../../../redux/slices/availability'

export const LAST_AVAILABILITY = gql`
  {
    lastAvailability {
      status
    }
  }
`

export const Availability = () => {
  const dispatch = useDispatch()
  const {loading, error, data} = useQuery(LAST_AVAILABILITY)
  const availability = useSelector(({ availability }) => availability)

  useEffect(() => {
    if (!loading) {
      if (error) {
        console.log('AVAILABILITY ERROR: ', error)
      }

      const loadedStatus = data?.lastAvailability?.status

      const statusTexts = {
        [AVAILABILITIES.available]: 'Available',
        [AVAILABILITIES.partially_available]: 'Partially available',
        [AVAILABILITIES.not_available]: 'Not available'
      }

      if (loadedStatus) {
        dispatch(setAvailability({
          status: loadedStatus,
          statusText: statusTexts[loadedStatus],
        }))
      }
    }
  }, [data, error, loading, dispatch])

  return (
    <AvailabilityContainer>
      <AvailabilityImage
        status={availability.status}
        loading={loading}
      />
      <AvailabilityText>
        <span>{availability.statusText}</span> for charity organization project
      </AvailabilityText>
    </AvailabilityContainer>
  )
}
