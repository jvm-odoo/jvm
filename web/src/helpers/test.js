import React from 'react'
import toJson from 'enzyme-to-json'
import { act } from 'react-dom/test-utils'
import { mount, shallow } from 'enzyme'
import { MockedProvider } from '@apollo/react-testing'
import { Provider } from 'react-redux'
import { ThemeProvider } from 'styled-components'

import { store } from '../redux/store'
import { THEME_LIGHT } from '../constants'

const wait = (time = 0) => new Promise(res => setTimeout(res, time))

export const executeMockProviderTestCase = async (wrapperInstance) => (
  await act(async () => {
    await wait(100)
    wrapperInstance.update()
  })
)

export const shallowMatchSnapshot = (Component) => {
  const wrapper = shallow(
    <ThemeProvider theme={THEME_LIGHT}>
      {Component}
    </ThemeProvider>)
  expect(toJson(wrapper)).toMatchSnapshot()
}

export const mountWrapperWithMockedData = async (Component, mocks) => {
  let wrapper

  act(() => {
    wrapper = mount(
      <MockedProvider mocks={mocks} addTypename={false}>
        <ThemeProvider theme={THEME_LIGHT}>
          {Component}
        </ThemeProvider>
      </MockedProvider>
    )
  })

  await executeMockProviderTestCase(wrapper)
  return wrapper
}

export const withReduxProvider = (Component) => (
  <Provider store={store}>
    {Component}
  </Provider>
)
