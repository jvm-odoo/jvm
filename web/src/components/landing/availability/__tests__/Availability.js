import React from 'react'
import toJson from 'enzyme-to-json'

import { Availability, LAST_AVAILABILITY } from '../Availability'
import { AVAILABILITIES } from '../../../../constants'
import { mountWrapperWithMockedData } from '../../../../helpers/test'

const generateMock = (status) => (
  [{
    request: {
      query: LAST_AVAILABILITY,
    },
    result: {
      data: { lastAvailability: { id: 1, status } },
    },
  }]
)

it('renders correctly when server down/loading', async () => {
  const wrapper = await mountWrapperWithMockedData(
    <Availability />,
    generateMock(AVAILABILITIES.loading)
  )

  expect(wrapper.find('p span').text()).toBe('Maybe available')
  expect(toJson(wrapper)).toMatchSnapshot()
})

it('renders correctly when available', async () => {
  const wrapper = await mountWrapperWithMockedData(
    <Availability />,
    generateMock(AVAILABILITIES.available)
  )

  expect(wrapper.find('p span').text()).toBe('Available')
})

it('renders correctly when server partially available', async () => {
  const wrapper = await mountWrapperWithMockedData(
    <Availability />,
    generateMock(AVAILABILITIES.partially_available)
  )

  expect(wrapper.find('p span').text()).toBe('Partially available')
})

it('renders correctly when server not available', async () => {
  const wrapper = await mountWrapperWithMockedData(
    <Availability />,
    generateMock(AVAILABILITIES.not_available)
  )

  expect(wrapper.find('p span').text()).toBe('Not available')
})
