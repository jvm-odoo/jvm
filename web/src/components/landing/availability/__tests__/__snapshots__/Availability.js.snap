// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when server down/loading 1`] = `
<MockedProvider
  addTypename={false}
  mocks={
    Array [
      Object {
        "request": Object {
          "query": Object {
            "definitions": Array [
              Object {
                "directives": Array [],
                "kind": "OperationDefinition",
                "name": undefined,
                "operation": "query",
                "selectionSet": Object {
                  "kind": "SelectionSet",
                  "selections": Array [
                    Object {
                      "alias": undefined,
                      "arguments": Array [],
                      "directives": Array [],
                      "kind": "Field",
                      "name": Object {
                        "kind": "Name",
                        "value": "lastAvailability",
                      },
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "status",
                            },
                            "selectionSet": undefined,
                          },
                        ],
                      },
                    },
                  ],
                },
                "variableDefinitions": Array [],
              },
            ],
            "kind": "Document",
            "loc": Object {
              "end": 51,
              "start": 0,
            },
          },
        },
        "result": Object {
          "data": Object {
            "lastAvailability": Object {
              "id": 1,
              "status": 0,
            },
          },
        },
      },
    ]
  }
>
  <ApolloProvider
    client={
      ApolloClient {
        "cache": InMemoryCache {
          "addTypename": false,
          "cacheKeyRoot": KeyTrie {
            "weak": WeakMap {},
            "weakness": true,
          },
          "config": Object {
            "addTypename": false,
            "dataIdFromObject": [Function],
            "fragmentMatcher": HeuristicFragmentMatcher {},
            "freezeResults": false,
            "resultCaching": true,
          },
          "data": DepTrackingCache {
            "data": Object {
              "$ROOT_QUERY.lastAvailability": Object {
                "status": 0,
              },
              "ROOT_QUERY": Object {
                "lastAvailability": Object {
                  "generated": true,
                  "id": "$ROOT_QUERY.lastAvailability",
                  "type": "id",
                  "typename": undefined,
                },
              },
            },
            "depend": [Function],
          },
          "maybeBroadcastWatch": [Function],
          "optimisticData": DepTrackingCache {
            "data": Object {
              "$ROOT_QUERY.lastAvailability": Object {
                "status": 0,
              },
              "ROOT_QUERY": Object {
                "lastAvailability": Object {
                  "generated": true,
                  "id": "$ROOT_QUERY.lastAvailability",
                  "type": "id",
                  "typename": undefined,
                },
              },
            },
            "depend": [Function],
          },
          "silenceBroadcast": false,
          "storeReader": StoreReader {
            "executeSelectionSet": [Function],
            "executeStoreQuery": [Function],
            "executeSubSelectedArray": [Function],
            "freezeResults": false,
          },
          "storeWriter": StoreWriter {},
          "typenameDocumentCache": Map {},
          "watches": Set {
            Object {
              "callback": [Function],
              "optimistic": true,
              "previousResult": [Function],
              "query": Object {
                "definitions": Array [
                  Object {
                    "directives": Array [],
                    "kind": "OperationDefinition",
                    "name": undefined,
                    "operation": "query",
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "lastAvailability",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "status",
                                },
                                "selectionSet": undefined,
                              },
                            ],
                          },
                        },
                      ],
                    },
                    "variableDefinitions": Array [],
                  },
                ],
                "kind": "Document",
                "loc": Object {
                  "end": 51,
                  "start": 0,
                },
              },
              "variables": Object {},
            },
          },
        },
        "clearStoreCallbacks": Array [],
        "defaultOptions": Object {},
        "disableNetworkFetches": false,
        "link": MockLink {
          "addTypename": false,
          "mockedResponsesByKey": Object {
            "{\\"query\\":\\"{\\\\n  lastAvailability {\\\\n    status\\\\n  }\\\\n}\\\\n\\"}": Array [],
          },
        },
        "localState": LocalState {
          "cache": InMemoryCache {
            "addTypename": false,
            "cacheKeyRoot": KeyTrie {
              "weak": WeakMap {},
              "weakness": true,
            },
            "config": Object {
              "addTypename": false,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "freezeResults": false,
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {
                "$ROOT_QUERY.lastAvailability": Object {
                  "status": 0,
                },
                "ROOT_QUERY": Object {
                  "lastAvailability": Object {
                    "generated": true,
                    "id": "$ROOT_QUERY.lastAvailability",
                    "type": "id",
                    "typename": undefined,
                  },
                },
              },
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {
                "$ROOT_QUERY.lastAvailability": Object {
                  "status": 0,
                },
                "ROOT_QUERY": Object {
                  "lastAvailability": Object {
                    "generated": true,
                    "id": "$ROOT_QUERY.lastAvailability",
                    "type": "id",
                    "typename": undefined,
                  },
                },
              },
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
              "executeSubSelectedArray": [Function],
              "freezeResults": false,
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {},
            "watches": Set {
              Object {
                "callback": [Function],
                "optimistic": true,
                "previousResult": [Function],
                "query": Object {
                  "definitions": Array [
                    Object {
                      "directives": Array [],
                      "kind": "OperationDefinition",
                      "name": undefined,
                      "operation": "query",
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "lastAvailability",
                            },
                            "selectionSet": Object {
                              "kind": "SelectionSet",
                              "selections": Array [
                                Object {
                                  "alias": undefined,
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "status",
                                  },
                                  "selectionSet": undefined,
                                },
                              ],
                            },
                          },
                        ],
                      },
                      "variableDefinitions": Array [],
                    },
                  ],
                  "kind": "Document",
                  "loc": Object {
                    "end": 51,
                    "start": 0,
                  },
                },
                "variables": Object {},
              },
            },
          },
          "client": [Circular],
        },
        "mutate": [Function],
        "query": [Function],
        "queryDeduplication": true,
        "queryManager": QueryManager {
          "assumeImmutableResults": false,
          "clientAwareness": Object {
            "name": undefined,
            "version": undefined,
          },
          "dataStore": DataStore {
            "cache": InMemoryCache {
              "addTypename": false,
              "cacheKeyRoot": KeyTrie {
                "weak": WeakMap {},
                "weakness": true,
              },
              "config": Object {
                "addTypename": false,
                "dataIdFromObject": [Function],
                "fragmentMatcher": HeuristicFragmentMatcher {},
                "freezeResults": false,
                "resultCaching": true,
              },
              "data": DepTrackingCache {
                "data": Object {
                  "$ROOT_QUERY.lastAvailability": Object {
                    "status": 0,
                  },
                  "ROOT_QUERY": Object {
                    "lastAvailability": Object {
                      "generated": true,
                      "id": "$ROOT_QUERY.lastAvailability",
                      "type": "id",
                      "typename": undefined,
                    },
                  },
                },
                "depend": [Function],
              },
              "maybeBroadcastWatch": [Function],
              "optimisticData": DepTrackingCache {
                "data": Object {
                  "$ROOT_QUERY.lastAvailability": Object {
                    "status": 0,
                  },
                  "ROOT_QUERY": Object {
                    "lastAvailability": Object {
                      "generated": true,
                      "id": "$ROOT_QUERY.lastAvailability",
                      "type": "id",
                      "typename": undefined,
                    },
                  },
                },
                "depend": [Function],
              },
              "silenceBroadcast": false,
              "storeReader": StoreReader {
                "executeSelectionSet": [Function],
                "executeStoreQuery": [Function],
                "executeSubSelectedArray": [Function],
                "freezeResults": false,
              },
              "storeWriter": StoreWriter {},
              "typenameDocumentCache": Map {},
              "watches": Set {
                Object {
                  "callback": [Function],
                  "optimistic": true,
                  "previousResult": [Function],
                  "query": Object {
                    "definitions": Array [
                      Object {
                        "directives": Array [],
                        "kind": "OperationDefinition",
                        "name": undefined,
                        "operation": "query",
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "lastAvailability",
                              },
                              "selectionSet": Object {
                                "kind": "SelectionSet",
                                "selections": Array [
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "status",
                                    },
                                    "selectionSet": undefined,
                                  },
                                ],
                              },
                            },
                          ],
                        },
                        "variableDefinitions": Array [],
                      },
                    ],
                    "kind": "Document",
                    "loc": Object {
                      "end": 51,
                      "start": 0,
                    },
                  },
                  "variables": Object {},
                },
              },
            },
          },
          "fetchQueryRejectFns": Map {},
          "idCounter": 3,
          "inFlightLinkObservables": Map {},
          "link": MockLink {
            "addTypename": false,
            "mockedResponsesByKey": Object {
              "{\\"query\\":\\"{\\\\n  lastAvailability {\\\\n    status\\\\n  }\\\\n}\\\\n\\"}": Array [],
            },
          },
          "localState": LocalState {
            "cache": InMemoryCache {
              "addTypename": false,
              "cacheKeyRoot": KeyTrie {
                "weak": WeakMap {},
                "weakness": true,
              },
              "config": Object {
                "addTypename": false,
                "dataIdFromObject": [Function],
                "fragmentMatcher": HeuristicFragmentMatcher {},
                "freezeResults": false,
                "resultCaching": true,
              },
              "data": DepTrackingCache {
                "data": Object {
                  "$ROOT_QUERY.lastAvailability": Object {
                    "status": 0,
                  },
                  "ROOT_QUERY": Object {
                    "lastAvailability": Object {
                      "generated": true,
                      "id": "$ROOT_QUERY.lastAvailability",
                      "type": "id",
                      "typename": undefined,
                    },
                  },
                },
                "depend": [Function],
              },
              "maybeBroadcastWatch": [Function],
              "optimisticData": DepTrackingCache {
                "data": Object {
                  "$ROOT_QUERY.lastAvailability": Object {
                    "status": 0,
                  },
                  "ROOT_QUERY": Object {
                    "lastAvailability": Object {
                      "generated": true,
                      "id": "$ROOT_QUERY.lastAvailability",
                      "type": "id",
                      "typename": undefined,
                    },
                  },
                },
                "depend": [Function],
              },
              "silenceBroadcast": false,
              "storeReader": StoreReader {
                "executeSelectionSet": [Function],
                "executeStoreQuery": [Function],
                "executeSubSelectedArray": [Function],
                "freezeResults": false,
              },
              "storeWriter": StoreWriter {},
              "typenameDocumentCache": Map {},
              "watches": Set {
                Object {
                  "callback": [Function],
                  "optimistic": true,
                  "previousResult": [Function],
                  "query": Object {
                    "definitions": Array [
                      Object {
                        "directives": Array [],
                        "kind": "OperationDefinition",
                        "name": undefined,
                        "operation": "query",
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "lastAvailability",
                              },
                              "selectionSet": Object {
                                "kind": "SelectionSet",
                                "selections": Array [
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "status",
                                    },
                                    "selectionSet": undefined,
                                  },
                                ],
                              },
                            },
                          ],
                        },
                        "variableDefinitions": Array [],
                      },
                    ],
                    "kind": "Document",
                    "loc": Object {
                      "end": 51,
                      "start": 0,
                    },
                  },
                  "variables": Object {},
                },
              },
            },
            "client": [Circular],
          },
          "mutationStore": MutationStore {
            "store": Object {},
          },
          "onBroadcast": [Function],
          "pollingInfoByQueryId": Map {},
          "queries": Map {
            "1" => Object {
              "cancel": [Function],
              "document": Object {
                "definitions": Array [
                  Object {
                    "directives": Array [],
                    "kind": "OperationDefinition",
                    "name": undefined,
                    "operation": "query",
                    "selectionSet": Object {
                      "kind": "SelectionSet",
                      "selections": Array [
                        Object {
                          "alias": undefined,
                          "arguments": Array [],
                          "directives": Array [],
                          "kind": "Field",
                          "name": Object {
                            "kind": "Name",
                            "value": "lastAvailability",
                          },
                          "selectionSet": Object {
                            "kind": "SelectionSet",
                            "selections": Array [
                              Object {
                                "alias": undefined,
                                "arguments": Array [],
                                "directives": Array [],
                                "kind": "Field",
                                "name": Object {
                                  "kind": "Name",
                                  "value": "status",
                                },
                                "selectionSet": undefined,
                              },
                            ],
                          },
                        },
                      ],
                    },
                    "variableDefinitions": Array [],
                  },
                ],
                "kind": "Document",
                "loc": Object {
                  "end": 51,
                  "start": 0,
                },
              },
              "invalidated": false,
              "lastRequestId": 2,
              "listeners": Set {
                [Function],
              },
              "newData": null,
              "observableQuery": ObservableQuery {
                "_subscriber": [Function],
                "isTornDown": false,
                "lastResult": Object {
                  "data": Object {
                    "lastAvailability": Object {
                      "status": 0,
                    },
                  },
                  "loading": false,
                  "networkStatus": 7,
                  "stale": false,
                },
                "lastResultSnapshot": Object {
                  "data": Object {
                    "lastAvailability": Object {
                      "status": 0,
                    },
                  },
                  "loading": false,
                  "networkStatus": 7,
                  "stale": false,
                },
                "observers": Set {
                  SubscriptionObserver {
                    "_subscription": Subscription {
                      "_cleanup": [Function],
                      "_observer": Object {
                        "error": [Function],
                        "next": [Function],
                      },
                      "_queue": undefined,
                      "_state": "ready",
                    },
                  },
                },
                "options": Object {
                  "context": undefined,
                  "displayName": "Query",
                  "metadata": Object {
                    "reactComponent": Object {
                      "displayName": "Query",
                    },
                  },
                  "notifyOnNetworkStatusChange": false,
                  "query": Object {
                    "definitions": Array [
                      Object {
                        "directives": Array [],
                        "kind": "OperationDefinition",
                        "name": undefined,
                        "operation": "query",
                        "selectionSet": Object {
                          "kind": "SelectionSet",
                          "selections": Array [
                            Object {
                              "alias": undefined,
                              "arguments": Array [],
                              "directives": Array [],
                              "kind": "Field",
                              "name": Object {
                                "kind": "Name",
                                "value": "lastAvailability",
                              },
                              "selectionSet": Object {
                                "kind": "SelectionSet",
                                "selections": Array [
                                  Object {
                                    "alias": undefined,
                                    "arguments": Array [],
                                    "directives": Array [],
                                    "kind": "Field",
                                    "name": Object {
                                      "kind": "Name",
                                      "value": "status",
                                    },
                                    "selectionSet": undefined,
                                  },
                                ],
                              },
                            },
                          ],
                        },
                        "variableDefinitions": Array [],
                      },
                    ],
                    "kind": "Document",
                    "loc": Object {
                      "end": 51,
                      "start": 0,
                    },
                  },
                  "variables": Object {},
                },
                "queryId": "1",
                "queryManager": [Circular],
                "queryName": undefined,
                "shouldSubscribe": true,
                "subscriptions": Set {},
                "variables": Object {},
              },
              "subscriptions": Set {},
            },
          },
          "queryDeduplication": true,
          "queryStore": QueryStore {
            "store": Object {
              "1": Object {
                "document": Object {
                  "definitions": Array [
                    Object {
                      "directives": Array [],
                      "kind": "OperationDefinition",
                      "name": undefined,
                      "operation": "query",
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "lastAvailability",
                            },
                            "selectionSet": Object {
                              "kind": "SelectionSet",
                              "selections": Array [
                                Object {
                                  "alias": undefined,
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "status",
                                  },
                                  "selectionSet": undefined,
                                },
                              ],
                            },
                          },
                        ],
                      },
                      "variableDefinitions": Array [],
                    },
                  ],
                  "kind": "Document",
                  "loc": Object {
                    "end": 51,
                    "start": 0,
                  },
                },
                "graphQLErrors": Array [],
                "metadata": Object {
                  "reactComponent": Object {
                    "displayName": "Query",
                  },
                },
                "networkError": null,
                "networkStatus": 7,
                "previousVariables": null,
                "variables": Object {},
              },
            },
          },
          "ssrMode": false,
          "transformCache": WeakMap {},
        },
        "reFetchObservableQueries": [Function],
        "resetStore": [Function],
        "resetStoreCallbacks": Array [],
        "store": DataStore {
          "cache": InMemoryCache {
            "addTypename": false,
            "cacheKeyRoot": KeyTrie {
              "weak": WeakMap {},
              "weakness": true,
            },
            "config": Object {
              "addTypename": false,
              "dataIdFromObject": [Function],
              "fragmentMatcher": HeuristicFragmentMatcher {},
              "freezeResults": false,
              "resultCaching": true,
            },
            "data": DepTrackingCache {
              "data": Object {
                "$ROOT_QUERY.lastAvailability": Object {
                  "status": 0,
                },
                "ROOT_QUERY": Object {
                  "lastAvailability": Object {
                    "generated": true,
                    "id": "$ROOT_QUERY.lastAvailability",
                    "type": "id",
                    "typename": undefined,
                  },
                },
              },
              "depend": [Function],
            },
            "maybeBroadcastWatch": [Function],
            "optimisticData": DepTrackingCache {
              "data": Object {
                "$ROOT_QUERY.lastAvailability": Object {
                  "status": 0,
                },
                "ROOT_QUERY": Object {
                  "lastAvailability": Object {
                    "generated": true,
                    "id": "$ROOT_QUERY.lastAvailability",
                    "type": "id",
                    "typename": undefined,
                  },
                },
              },
              "depend": [Function],
            },
            "silenceBroadcast": false,
            "storeReader": StoreReader {
              "executeSelectionSet": [Function],
              "executeStoreQuery": [Function],
              "executeSubSelectedArray": [Function],
              "freezeResults": false,
            },
            "storeWriter": StoreWriter {},
            "typenameDocumentCache": Map {},
            "watches": Set {
              Object {
                "callback": [Function],
                "optimistic": true,
                "previousResult": [Function],
                "query": Object {
                  "definitions": Array [
                    Object {
                      "directives": Array [],
                      "kind": "OperationDefinition",
                      "name": undefined,
                      "operation": "query",
                      "selectionSet": Object {
                        "kind": "SelectionSet",
                        "selections": Array [
                          Object {
                            "alias": undefined,
                            "arguments": Array [],
                            "directives": Array [],
                            "kind": "Field",
                            "name": Object {
                              "kind": "Name",
                              "value": "lastAvailability",
                            },
                            "selectionSet": Object {
                              "kind": "SelectionSet",
                              "selections": Array [
                                Object {
                                  "alias": undefined,
                                  "arguments": Array [],
                                  "directives": Array [],
                                  "kind": "Field",
                                  "name": Object {
                                    "kind": "Name",
                                    "value": "status",
                                  },
                                  "selectionSet": undefined,
                                },
                              ],
                            },
                          },
                        ],
                      },
                      "variableDefinitions": Array [],
                    },
                  ],
                  "kind": "Document",
                  "loc": Object {
                    "end": 51,
                    "start": 0,
                  },
                },
                "variables": Object {},
              },
            },
          },
        },
        "typeDefs": undefined,
        "version": "2.6.10",
        "watchQuery": [Function],
      }
    }
  >
    <Provider
      store={
        Object {
          "dispatch": [Function],
          "getState": [Function],
          "replaceReducer": [Function],
          "subscribe": [Function],
          Symbol(observable): [Function],
        }
      }
    >
      <Availability>
        <styled.div>
          <div
            className="sc-AxjAm iZilVu"
          >
            <AvailabilityImage
              loading={false}
              status={0}
            >
              <styled.div>
                <div
                  className="sc-AxirZ gpWHsK"
                >
                  <styled.img
                    alt="Jason Van Malder"
                    src="me.jpg"
                  >
                    <img
                      alt="Jason Van Malder"
                      className="sc-AxiKw jKjsTW"
                      src="me.jpg"
                    />
                  </styled.img>
                  <styled.span
                    status={0}
                  >
                    <span
                      className="sc-AxhCb gFhkBz"
                    />
                  </styled.span>
                </div>
              </styled.div>
            </AvailabilityImage>
            <styled.p>
              <p
                className="sc-AxhUy gWGACW"
              >
                <span>
                  Maybe available
                </span>
                 for charity organization project
              </p>
            </styled.p>
          </div>
        </styled.div>
      </Availability>
    </Provider>
  </ApolloProvider>
</MockedProvider>
`;
